src/lib/loaders/batchLoader.ts

```diff
@@ -20,35 +62,35 @@ const renderParams = key => {
     .join("&")
 }
 
-interface GroupKeysResult {
-  id: string[]
-  size: number
-  [key: string]: any
-}
-export const groupKeys = (
-  requestedKeys: string | { id }
-): GroupKeysResult[] => {
-  const grouped = chain(requestedKeys)
-    .groupBy(renderParams)
-    .values()
-    .map(values => chunk(values, 20))
-    .flatten()
-    .map((keys: string[] | { id }[]) => {
-      if (typeof keys[0] === "string") {
-        return { id: keys, size: keys.length }
-      }
-      return {
-        // @ts-ignore
-        ...keys[0],
-        id: keys.map(k => k.id),
-        size: keys.length,
+/**
+ * Takes an array of normalized keys and groups them based on their parameters.
+ * @returns a tuple of an array of group strings and an array of grouped keys
+ */
+export const groupKeys = (requestedKeys): [KeyGroupList, NormalKey[][]] => {
+  const [keyGroupList, groupedKeys] = chain(requestedKeys)
+    .groupBy(getKeyGroup)
+    .entries()
+    .thru(entries => {
+      let result: [KeyGroupList, NormalKey[][]] = [[], []]
```

Comment:
Pretty sure mutating the arrays doesn't require `let`. i.e. the pointers to the arrays remain the same.

```suggestion
      const result: [KeyGroupList, NormalKey[][]] = [[], []]
```